### ISSUE TITLE
Replicable Hash Collisions (2)

### ISSUE THREAD

**Issue by zkldi** [fraunhoferfokus/JSum#7]
Sorry to bother again - I've found another way to cause hash collisions.

```js
serialise({a: 1, b: 2});
// a:1|b:2

serialise({"a:1|b": 2})
// a:1|b:2
```

Assumedly the same fix could be done by wrapping K in quotes. Can write a pull request for that if you want?

**Comment by yan-foto** [comment]
It's no bother at all! Fixing stuff is always welcome. But I don't suppose that your solution will work (I'll give you another day and you'll open another issue here). The proper way would be to use a character, e.g., a control character, that is not going to be valid in any case. I know for a fact, that `\u0000` (the null character)  is going to give you `Uncaught SyntaxError: Unexpected token  in JSON ` if you try to parse it. This is not necessarily the case for JS objects (remember JS Obj != JSON).

Long story short, if you want to send a PR start with:

```js
const DELIM = '\u0000'
```

and adapt the tests respectively. Please also add the cases that you mention in this issue to the tests. That would be a great help!

**Comment by yan-foto** [comment]
This should solve the problem.
